airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -311,20 +311,32 @@ def _get_web_identity_credential_fetcher(
     ) -> botocore.credentials.AssumeRoleWithWebIdentityCredentialFetcher:
         base_session = self.basic_session._session or botocore.session.get_session()
         client_creator = base_session.create_client
-        federation = self.extra_config.get("assume_role_with_web_identity_federation")
-        if federation == "google":
-            web_identity_token_loader = self._get_google_identity_token_loader()
-        else:
-            raise AirflowException(
-                f'Unsupported federation: {federation}. Currently "google" only are supported.'
-            )
+        federation = str(self.extra_config.get("assume_role_with_web_identity_federation"))
+
+        web_identity_token_loader = {
+            "file": self._get_file_token_loader,
+            "google": self._get_google_identity_token_loader,
+        }.get(federation, lambda: None)()
+
+        if not web_identity_token_loader:
+            raise AirflowException(f"Unsupported federation: {federation}.")
+
         return botocore.credentials.AssumeRoleWithWebIdentityCredentialFetcher(
             client_creator=client_creator,
             web_identity_token_loader=web_identity_token_loader,
```

Comment:
This "default" lambda pattern is a bit odd/unusual, so lets do this instead

```suggestion
        web_identity_token_loader = {
            "file": self._get_file_token_loader,
            "google": self._get_google_identity_token_loader,
        }.get(federation)

        if not web_identity_token_loader:
            raise AirflowException(f"Unsupported federation: {federation}.")

        return botocore.credentials.AssumeRoleWithWebIdentityCredentialFetcher(
            client_creator=client_creator,
            web_identity_token_loader=web_identity_token_loader(),
```

Reply:
Agreed, this is better.